synapse/http/servlet.py

```diff
@@ -191,6 +222,59 @@ def parse_string_from_args(
             return default
 
 
+def parse_string_from_args(
+    args: List[str],
+    name: Union[bytes, str],
+    default: Optional[str] = None,
+    required: bool = False,
+    allowed_values: Optional[Iterable[str]] = None,
+    encoding: Optional[str] = "ascii",
+) -> Optional[Union[bytes, str]]:
+    """
+    Parse the string parameter from the request query string list
+    and return the first result.
+
+    If encoding is not None, the content of the query param will be
+    decoded to Unicode using the encoding, otherwise it will be encoded
+
+    Args:
+        args (List[str]): the twisted HTTP request.args list.
+        name (bytes|unicode): the name of the query parameter.
+        default (bytes|unicode|None): value to use if the parameter is absent,
+            defaults to None. Must be bytes if encoding is None.
+        required (bool): whether to raise a 400 SynapseError if the
+            parameter is absent, defaults to False.
+        allowed_values (list[bytes|unicode]): List of allowed values for the
+            string, or None if any value is allowed, defaults to None. Must be
+            the same type as name, if given.
+        encoding (str|None): The encoding to decode the string content with.
+
+    Returns:
+        bytes/unicode|None: A string value or the default. Unicode if encoding
+        was given, bytes otherwise.
+
+    Raises:
+        SynapseError if the parameter is absent and required, or if the
+            parameter is present, must be one of a list of allowed values and
+            is not one of those allowed values.
+    """
+
+    strings = parse_strings_from_args(
+        args,
+        name,
+        default=default,
+        required=required,
+        allowed_values=allowed_values,
+        encoding=encoding,
+    )
+
+    if isinstance(strings, list) and len(strings):
```

Comment:
I think this is attempt to differentiate between `None` and an empty list or something? This is confusing me a bit. Is it possible for `strings` to be an empty list? If so, then this function will return `[]` which doesn't match the types / comments.

Reply:
hmm, I see what you're saying. Only `isinstance(strings, list)` is necessary here here because the list will be populated or defaulted.

Originally, I added `isinstance(strings, list)` is to avoid calling `len(None)` but that's just extra now.